### PR TITLE
implement /projects/:id/publish and /projects/:id/unpublish routes

### PR DESCRIPTION
**Pull request by ashleygwilliams**

### CODE REVIEW

File: api/classes/publish.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var AWS = require('aws-sdk'); 
+var s3 = new AWS.S3(); 
+
+var isPublished = function(project){
+  return project.get('publish_url') !== null;
+};
+
+exports.publish = function(project) {
+  var url = "FAKE://URL";
```
Comment (sedge) [inline api/classes/publish.js]: I assume this is where the s3 logic will go when the time comes?

File: api/classes/publish.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+exports.unpublish = function(project) {
+  var message = "";
+  if (isPublished(project)) {
+    project.set({ publish_url: null }).save();
+    message =  "Successfully unpublished " + project.get('title');
+  } else {
+    message = "Cannot unpublish an project that is not published.";
+  }
+  console.log(message);
```
Comment (sedge) [inline api/classes/publish.js]: nuke this sucker

File: api/classes/publish.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+var s3 = new AWS.S3(); 
+
+var isPublished = function(project){
+  return project.get('publish_url') !== null;
+};
+
+exports.publish = function(project) {
+  var url = "FAKE://URL";
+  project.set({ publish_url:  url }).save();
+  console.log("Successfully published " +
```
Comment (sedge) [inline api/classes/publish.js]: We have a logger don't we? Let's use that instead.

File: api/classes/publish.js (hunk 1 above)
Comment (ashleygwilliams) [inline api/classes/publish.js]: indeed.

File: api/classes/publish.js (hunk 3 above)
Comment (ashleygwilliams) [inline api/classes/publish.js]: oh right. lol. yes, that makes sense.

File: test/api/handlers/projects/publish.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+var Lab = require('lab');
+var lab = exports.lab = Lab.script();
+
+var experiment = lab.experiment;
+var test = lab.test;
+var before = lab.before;
+var after = lab.after;
+var expect = require('code').expect;
+
+var config = require("../../../lib/fixtures/files").update;
```
Comment (sedge) [inline test/api/handlers/projects/publish.js]: This is supposed to contain the fail or success options being passed to the server in each test. This is pulling from the `files` fixtures, so these tests will behave strangely.

File: test/api/handlers/projects/unpublish.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+var Lab = require('lab');
+var lab = exports.lab = Lab.script();
+
+var experiment = lab.experiment;
+var test = lab.test;
+var before = lab.before;
+var after = lab.after;
+var expect = require('code').expect;
+
+var config = require("../../../lib/fixtures/files").update;
```
Comment (sedge) [inline test/api/handlers/projects/unpublish.js]: ditto.

**Pull request merged**